httpbakery/form/form.go

```diff
@@ -123,9 +140,21 @@ func (v webPageVisitor) visitWebPage(u *url.URL) error {
 	req.Header.Set("Accept", "application/json")
 	var lm loginMethods
 	if err := v.client.Do(req, nil, &lm); err != nil {
+		if v.fallback != nil {
+			if err := v.fallback(u); err != nil {
+				return errgo.NoteMask(err, "form login not supported, fallback failed", errgo.Any)
```

Comment:
I think you should just return errgo.Mask(err, errgo.Any) here. We're setting up a fallback chain here, and I don't think that each link in the chain should prefix the error message from the next fallback.